src/Tasks/MSBuild.cs

```diff
@@ -289,7 +292,21 @@ public override bool Execute()
                     break;
                 }
 
-                if (FileSystems.Default.FileExists(projectPath) || (_skipNonexistentProjects == SkipNonexistentProjectsBehavior.Build))
+                // Try to get the behavior from metadata if it is undefined.
+                var skipNonexistPropjects = _skipNonexistentProjects;
```

Comment:
It looks like this was essentially copied from SolutionProjectGenerator—is it possible to merge them? Are there major differences? Why are they separate in the first place?

Reply:
I don't know why there are two copies of MSBuild.cs and MSBuild_Tests.cs.

When tests are running, they resolve to BackEnd.